### PR TITLE
SwapFactory to SwapCreator rename, upgrade to solc 0.8.19

### PR DESCRIPTION
**Pull request by dimalinux**
Changes:
* Renames SwapFactory to SwapCreator
* Upgrades from solc 0.8.17 to 0.8.19
* Updates stagenet/Sepolia contract addresses
* Harmonizing naming of the address variables for the contracts
* Provides github secrets support for an `ETH_SEPOLIA_ENDPOINT` environment variable used by unit tests validating the deployed contract

### CODE REVIEW

File: docs/developing.md [hunk 1]
```diff
@@ -12,19 +12,28 @@ You can use the script `scripts/setup-env.sh` to quickly set up local monerod-te
 ganache instances. If you need to later kill `ganache`, `monerod`, `monero-wallet-rpc`, or
 `swapd` instances, you can use `scripts/cleanup-test-processes.sh`.
 
-## Deploying or using deployed SwapFactory.sol
-
-The swap program uses a "factory" contract for the Ethereum side to reduce gas costs from deploying a new contract for each swap. The contract can be found in [here](../ethereum/contracts/SwapFactory.sol). For each new swap, the eth-holding party will call `NewSwap` on the factory contract, initiating a swap instance inside the contract.
```
Comment (dimalinux) [inline docs/developing.md:17]: Other than "factory" => "creator" substitution, no other content changes were made in this file. Added newlines are causing the larger diff. This was a 340 character long line before.

File: .github/workflows/unit-tests.yml [hunk 2]
```diff
@@ -59,6 +59,7 @@ jobs:
       - name: Run unit tests
         env:
           ETH_MAINNET_ENDPOINT: ${{ secrets.ETH_MAINNET_ENDPOINT }}
+          ETH_SEPOLIA_ENDPOINT: ${{ secrets.ETH_SEPOLIA_ENDPOINT }}
```
Comment (dimalinux) [inline .github/workflows/unit-tests.yml:62]: This is all setup if you add a github actions secret for it. The stagenet contract check tests will happen even if it is not set, but they will use `https://rpc.sepolia.org/`, which may or may not be reliable enough.

**Comment by dimalinux** [comment]
The -2.29% project-wide code-coverage stat is some kind of bug. If you look at the impacted files, there is nothing that can explain where it was lost.

File: .github/workflows/unit-tests.yml (hunk 2 above)
Comment (noot) [reply .github/workflows/unit-tests.yml:62]: added

File: net/host.go [hunk 3]
```diff
@@ -26,7 +26,7 @@ import (
 const (
 	// ProtocolID is the base atomic swap network protocol ID prefix. The full ID
 	// includes the chain ID at the end.
-	ProtocolID          = "/atomic-swap/0.2"
+	ProtocolID          = "/atomic-swap/0.3"
```
Comment (noot) [inline net/host.go:29]: is this in anticipation of the release? :p

File: protocol/backend/backend.go [hunk 4]
```diff
@@ -100,34 +100,34 @@ type backend struct {
 	perSwapXMRDepositAddrRWMu sync.RWMutex
 	perSwapXMRDepositAddr     map[types.Hash]*mcrypto.Address
 
-	// swap contract
-	contract     *contracts.SwapFactory
-	contractAddr ethcommon.Address
-	swapTimeout  time.Duration
+	// swap swapCreator
```
Comment (noot) [inline protocol/backend/backend.go:103]: swap contract?

**Review by noot — APPROVED** [review]
looks good!

File: net/host.go (hunk 3 above)
Comment (dimalinux) [reply net/host.go:29]: Yes, but going forward we need to update the protocol version every time p2p field names change or when the SwapCreator contract bytecode changes (and this PR does both).

File: protocol/backend/backend.go (hunk 4 above)
Comment (dimalinux) [reply protocol/backend/backend.go]: I updated it to "swap contract".

**Pull request merged**